File: easy/base_ball/Go/base_ball.go
package main

// Cathegory: Easy
// Success
// Details
// Runtime: 6 ms, faster than 31.51% of Go online submissions for Baseball Game.
// Memory Usage: 2.8 MB, less than 15.07% of Go online submissions for Baseball Game.

// 39 / 39 test cases passed.
// 	Status: Accepted
// Runtime: 6 ms
// Memory Usage: 2.8 MB

// copyright 2022 Bogdan Peta 

import (
	// "fmt"
	"strconv"
)

func calcPoints(input []string) int {

	var out []int = []int{}

	for _, ch := range input {
		val, err := strconv.Atoi(string(ch))

		if err == nil {
			out = append(out, val)
		} else {
			if ch == "D" {
				crt := out[len(out) - 1]
				out = out[0:len(out) - 1]
				out = append(out, crt, crt * 2)
			}
			if ch == "C" {
				out = out[0:len(out) - 1]
			}
			if ch == "+" {
				x := out[len(out)-2:len(out)]
				out = out[0:len(out) - 2]
				out = append(out, x[0], x[1], x[0] + x[1])
			}
		}
	}
	
	sum := 0
	for _, elem  := range out {
		sum += elem
	}

	return sum
}

File: easy/base_ball/Rust/src/base_ball.rs
// Success
// Details
// Runtime: 0 ms, faster than 100.00% of Rust online submissions for Baseball Game.
// Memory Usage: 2.3 MB, less than 12.00% of Rust online submissions for Baseball Game.

// 39 / 39 test cases passed.
// 	Status: Accepted
// Runtime: 0 ms
// Memory Usage: 2.3 MB

// Copyright 2022 Bogdan Peta

pub fn calc_points(operations: Vec<String>) -> i32 
{
    let mut out: Vec<i32> = vec![];

    for ch in operations {
        let val0 = ch.parse::<i32>();

        match val0 {
            Ok(val) => {
                out.push(val);
            },
            Err(_e) => {
                if ch == "D" {
                    let crt = out.pop().unwrap();
                    out.push(crt);
                    out.push(crt * 2);
                } else if ch == "C" {
                    out.pop().unwrap();
                } else if ch == "+"{
                    let x1 = out.pop().unwrap();
                    let x0 = out.pop().unwrap();
                    out.push(x0);
                    out.push(x1);
                    out.push(x0 + x1);
                }
            },
        }
    }

    let mut sum = 0;

    for elem in out {
        sum += elem;
    }

    sum
}
